content/D_AlgebraicUnderpinnings.md

```diff
@@ -0,0 +1,600 @@
+
+## So much for theory
+
+Now we've reviewed the theoretical concepts behind LINQ, let's step back and look at how we use them. We have three kinds of operations:
+
+* Anamorphisms enter the sequence: `T1 --> IObservable<T2>`
+* Bind modifies the sequence. `IObservable<T1> --> IObservable<T2>`
+* Catamorphisms leave the sequence. Logically `IObservable<T1> --> T2`, but in practice typically `IObservable<T1> --> IObservable<T2>` where the output observable produces just a single value
+
+As an aside, bind and catamorphism were made famous by Google's [MapReduce](http://en.wikipedia.org/wiki/MapReduce) framework from Google. Here Google, refer to Bind and Catamorphism by names more commonly used in some functional languages, Map and Reduce.
+
+Most Rx operators are actually specializations of the higher order functional concepts. To give a few examples:
+
+- Anamorphisms:
+  - [`Generate`](03_CreatingObservableSequences.md#observablegenerate)
+  - [`Range`](03_CreatingObservableSequences.md#observablerange)
+  - [`Return`](03_CreatingObservableSequences.md#observablereturn)
+- Bind:
+  - [`SelectMany`](06_Transformation.md#selectmany)
+  - [`Select`](06_Transformation.md#select)
+  - [`Where`](05_Filtering.md)
+- Catamorphism:
+  - [`Aggregate`](07_Aggregation.md#aggregate)
+  - [`Sum`](07_Aggregation.md#sum)
+  - [`Min` and `Max`](07_Aggregation.md#min-and-max)
+
+
+## Amb
+
+The `Amb` method was a new concept to me when I started using Rx. This function was first introduced by [John McCarthy](https://en.wikipedia.org/wiki/John_McCarthy_(computer_scientist)), in his 1961 paper ['A Basis for a Mathematical Theory of Computation'](https://www.cambridge.org/core/journals/journal-of-symbolic-logic/article/abs/john-mccarthy-a-basis-for-a-mathematical-theory-of-computation-preliminary-report-proceedings-of-the-western-joint-computer-conference-papers-presented-at-the-joint-ireaieeacm-computer-conference-los-angeles-calif-may-911-1961-western-joint-computer-conference-1961-pp-225238-john-mccarthy-a-basis-for-a-mathematical-theory-of-computation-computer-programming-and-formal-systems-edited-by-p-braffort-and-d-hirschberg-studies-in-logic-and-the-foundations-of-mathematics-northholland-publishing-company-amsterdam1963-pp-3370/D1AD4E0CDB7FBE099B04BB4DAF24AFFA) in the Proceedings of the Western Joint Computer Conference. (A digital copy of this is hard to find, but a later version was published in [1963](http://www-formal.stanford.edu/jmc/basis1.pdf) in 'Computer Programming and Format Systems'.) It is an abbreviation of the word _Ambiguous_. Rx diverges slightly from normal .NET class library naming conventions here in using this abbreviation, partly because `amb` is the established name for this operator, but also as a tribute to McCarthy, whose work was an inspiration for the design of Rx.
```

Comment:
> The `Amb` method was a new concept to me when I started using Rx

Is this from Lee rather than Ian?

Reply:
Lee wrote it, but it happens to be an entirely accurate statement about Ian too, so I left it in.